test/README.md

```diff
@@ -0,0 +1,102 @@
+# Writing Tests
+
+There are multiple flavors of node-addon-api test build that covers different
+build flags defined in `napi.h`:
+
+1. c++ exceptions enabled,
+2. c++ exceptions disabled,
+3. c++ exceptions disabled, and `NODE_ADDON_API_ENABLE_MAYBE` is defined.
+
+`Napi` functions that calling into JavaScript can have different declared
+`NODE_ADDON_API_ENABLE_MAYBE` is defined and c++ exceptions disabled
+return types to reflect build flavor settings. For example,
+`Napi::Object::Set` returns `bool`, and `Napi::Maybe<bool>` when
+`NODE_ADDON_API_ENABLE_MAYBE` is defined. In source code, return type variants
+are defined as `Napi::MaybeOrValue<>` to prevent from duplicating most of the
+code base.
+
+To properly test these build flavors, we should take care of the return value
+of `Napi` functions that may call into JavaScript. For example:
+
+```cpp
+#include "napi.h"
+#include "test_helper.h"
+
+using namespace Napi;
+
+Value fn(const CallbackInfo& info) {
+  Object obj = info[0].As<Object>();
+  Value value = MaybeUnwrap(obj->Get("foobar")); // <- `obj->Get` may throws
```

Comment:
Should this be in a return or is it that we need to use the helpers for more than returns as well?